tests/filecheck/transforms/copy_to_dma.mlir

```diff
@@ -64,47 +64,51 @@
+// CHECK-NEXT:     %6 = "arith.constant"() <{"value" = 2 : index}> : () -> index
+// CHECK-NEXT:     %7 = "arith.constant"() <{"value" = 4 : index}> : () -> index
+// CHECK-NEXT:     %8 = "arith.constant"() <{"value" = 2 : index}> : () -> index
+// CHECK-NEXT:     %9 = "arith.constant"() <{"value" = 4 : index}> : () -> index
+// CHECK-NEXT:     %10 = "arith.constant"() <{"value" = 128 : index}> : () -> index
+// CHECK-NEXT:     %11 = "arith.constant"() <{"value" = 32 : index}> : () -> index
+// CHECK-NEXT:     %12 = "arith.constant"() <{"value" = 128 : index}> : () -> index
+// CHECK-NEXT:     %13 = "arith.constant"() <{"value" = 4 : index}> : () -> index
+// CHECK-NEXT:     %14 = "arith.constant"() <{"value" = 16 : index}> : () -> index
+// CHECK-NEXT:     %15 = "arith.constant"() <{"value" = 128 : index}> : () -> index
+// CHECK-NEXT:     %16 = "arith.constant"() <{"value" = 16 : index}> : () -> index
+// CHECK-NEXT:     %17 = "arith.constant"() <{"value" = 128 : index}> : () -> index
+// CHECK-NEXT:     %18 = "arith.constant"() <{"value" = 4 : index}> : () -> index
+// CHECK-NEXT:     %19 = "arith.constant"() <{"value" = 64 : index}> : () -> index
+// CHECK-NEXT:     %20 = "arith.constant"() <{"value" = 16 : index}> : () -> index
+// CHECK-NEXT:     %21 = "arith.constant"() <{"value" = 0 : index}> : () -> index
+// CHECK-NEXT:     %22 = "arith.constant"() <{"value" = 1 : index}> : () -> index
+// CHECK-NEXT:     "scf.for"(%21, %6, %22) ({
+// CHECK-NEXT:     ^1(%23 : index):
+// CHECK-NEXT:       %24 = "arith.muli"(%23, %14) : (index, index) -> index
+// CHECK-NEXT:       %25 = "arith.addi"(%0, %24) : (index, index) -> index
+// CHECK-NEXT:       %26 = "arith.muli"(%23, %19) : (index, index) -> index
+// CHECK-NEXT:       %27 = "arith.addi"(%1, %26) : (index, index) -> index
+// CHECK-NEXT:       "scf.for"(%21, %8, %22) ({
+// CHECK-NEXT:       ^2(%28 : index):
+// CHECK-NEXT:         %29 = "arith.muli"(%28, %12) : (index, index) -> index
+// CHECK-NEXT:         %30 = "arith.addi"(%25, %29) : (index, index) -> index
+// CHECK-NEXT:         %31 = "arith.muli"(%28, %17) : (index, index) -> index
+// CHECK-NEXT:         %32 = "arith.addi"(%27, %31) : (index, index) -> index
+// CHECK-NEXT:         "func.call"(%30, %32, %20, %11, %16, %9) <{"callee" = @snax_dma_2d_transfer}> : (index, index, index, index, index, index) -> ()
+// CHECK-NEXT:         "scf.yield"() : () -> ()
+// CHECK-NEXT:       }) : (index, index, index) -> ()
+// CHECK-NEXT:       "scf.yield"() : () -> ()
+// CHECK-NEXT:     }) : (index, index, index) -> ()
+// CHECK-NEXT:     "func.return"() : () -> ()
+// CHECK-NEXT:   }) : () -> ()
+// CHECK-NEXT:   "func.func"() <{"sym_name" = "snax_dma_2d_transfer", "function_type" = (index, index, index, index, index, index) -> (), "sym_visibility" = "private"}> ({
+// CHECK-NEXT:   }) : () -> ()
+// CHECK-NEXT: }) : () -> ()
```

Comment:
The only change in the test result is that due to refactoring 4 extra ops are added (%2-%5). They are however not used and will be canonicalized away

Reply:
Can we, at some point, test whether these get canonicalized by mlir-opt in this repo?